### DISCUSSION TITLE
[Tables] Some row borders on Chrome and Edge are thicker than others

### DISCUSSION THREAD

**Discussion by the-overdriven** [tailwindlabs/tailwindcss#9595] [Help]
Some row borders on Chrome and Edge are thicker than others. Bug doesn't appear on Firefox. One can even see it on any table in your [docs](https://tailwindcss.com/docs/max-width), i.e:
![image](https://user-images.githubusercontent.com/100090726/196209807-d23cee61-7af1-4d7b-964c-f1e012122457.png)
Notice the border between max-w-7xl and max-w-full.

It seems to be coming from following property:
```
.divide-y > :not([hidden]) ~ :not([hidden]) {
    --tw-divide-y-reverse: 0;
    border-top-width: calc(1px * calc(1 - var(--tw-divide-y-reverse)));
```

Happens only on 100% and 110% zoom size.

System: Windows 10
Resolution: 1920x1080, browser window maximized

**Comment by thecrypticace** [comment]
1. What is your zoom set to in Windows itself?
2. Are you using a non-native resolution of the display? (Windows should say "(recommended)" next to it in Display Settings if it's a native resolution)
3. What make/model is your display?
4. Is it High DPI?

**Comment by the-overdriven** [comment]
@thecrypticace 
1. 100%
2. recommended
3. Graphics: GTX 1650, Screen: C49HG90DMU
4. 80 dpi?

**Reply by thecrypticace** [reply]
The C49HG90DMU appears to be a 49" 4K monitor with a native resolution of 3480x1080. Theoretically a resolution of 1920x1080 should not be scaled vertically — only horizontally. Given that you're seeing inconsistent results this might mean that the monitor itself is scaling the picture. Especially if Windows thinks 1920x1080 is the native resolution.

In PBP mode the "native" resolution would be presented as 1920x1080 to the computer. Are you using it in picture-by-picture mode? Do you see any difference if you switch that off?

**Reply by the-overdriven** [reply]
Yes, I use PBP, but it's the same without it. Must be a quirky bug with calc function in Chrome/Edge, because I don't see the thicker border on Firefox. Why not set the border width to just 0.5 px anyway? These calculations in CSS seem over-engineered. Classic excess of form over substance, common in frontend.

**Reply by adamwathan** [reply]
> Why not set the border width to just 0.5 px anyway?

Because it's supposed to be 1px, not 0.5px. The `calc` is there to support situations where you are using `flex-col-reverse` and the border needs to go on the bottom instead of the top. We only ever multiply it by `1` or `0` — `1 * 1px = 1px` and `0 * 1px = 0`, so it's just about toggling whether the border is on the top or the bottom.

Don't have a lot of patience for people who don't even attempt to really read the code or understand the problem talking down to the maintainers of the project like we don't know what we're doing — doesn't make me want to volunteer my time to help you.

**Reply by the-overdriven** [reply]
Thanks for explaining, but still to me it's just some flex-col-reverse edge case that's creating unnecessary noise in the code. As per rule: If you have to explain the code it's not a clean code.

Please don't get defensive, I'm merely reporting an issue with your framework, so I'm helping you, not the other way around.

**Reply by babakfp** [reply]
As far as I know, this is something related to Chromium. It happens on Chrome and Brave (because they use Chromium), but not on Firefox because Firefox doesn't use Chromium. If you **don't** use `calc` you can see **the issue still remains**.
